lamkey_nick_ndvi.py

```diff
@@ -0,0 +1,723 @@
+else:
+    print("Output directory does not exist, creating directory")
+    os.mkdir(os.path.join(working_dir, "ndvi-automation", "outputs"))
+
+
+# In[3]:
+
+
+# DO NOT MODIFY THIS CELL
+# Tests that the working directory is set to earth-analytics/data
+
+path = os.path.normpath(os.getcwd())
+student_wd_parts = path.split(os.sep)
+
+if student_wd_parts[-2:] == ['earth-analytics', 'data']:
+    print("\u2705 Great - it looks like your working directory is set correctly to ~/earth-analytics/data")
+else:
+    print("\u274C Oops, the autograder will not run unless your working directory is set to earth-analytics/data")
+
+
+# # Figure 1: Plot 1 - Mean NDVI For Each Site Across the Year (50 points)
+# 
+# Create a plot of the mean normalized difference vegetation index (NDVI) for the two different sites in the United States across the year: 
+# 
+# * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object.
+# * Each site should be identified with a different color in the plot and legend.
+# * The final plot **data should be cleaned to remove the influence of clouds**.
+# * Be sure to include appropriate title and axes labels.
+# 
+# Add additional cells as needed for processing data (e.g. defining functions, etc), but be sure to:
+# * follow the instructions in the code cells that have been provided to ensure that you are able to use the sanity check tests that are provided. 
+# * include only the plot code in the cell identified for the final plot code below
+
+# ## Task 1: 
+# 
+# In the cell below, create a single dataframe containing MEAN NDVI, the site name, 
+# and the date of the data for the HARV site 
+# scene `HARV/landsat-crop/LC080130302017031701T1-SC20181023151837`.  The column names for the  final
+# DataFrame should be`mean_ndvi`, and `site`, and the data should be **indexed on the date**. 
+# 
+# Use the functions that we reviewed in class (or create your own versions of them) to implement your code
+# 
+# ### In the Cell below Place  All Functions Needed to Run this Notebook (20 points)
+
+# In[4]:
+
+
+### DO NOT REMOVE THIS LINE OR EDIT / MOVE THIS CELL ###
+start_time = datetime.now()
+
+
+# In[5]:
+
+
+# In this cell place all of the functions needed to run your notebook
+# You will be graded here on function application, docstrings, efficiency so ensure
+# All functions are placed here!
+def open_clean_bands(band_path,
+                     crop_extent,
+                     valid_range=None):
+    """Open and mask a single landsat band using a pixel_qa layer.
+
+    Parameters
+    -----------
+    band_path : string
+        A path to the array to be opened
+    crop_extent : GeoDataFrame
+        A shapefile of AOI extents
+    valid_range : tuple (optional)
+        A tuple of min and max range of values for the data. Default = None
+
+
+    Returns
+    -----------
+    arr : xarray DataArray
+        An xarray DataArray with values that should be masked set to 1 for True (Boolean) 
```

Comment:
@nlamkey unneeded whitespace b/w these lines (at least this appears odd within the context of the github "inline review")